### PR TITLE
change(client): qtarget + ped turn face to vehicle entity

### PR DESCRIPTION
**Pull request by DoluTattoo**
Funny fact: If you're in a vehicle, your player will automaticly leave the vehicle and start refueling :)

### CODE REVIEW

File: client.lua [hunk 1]
```diff
@@ -179,6 +179,30 @@ local function StartFueling(vehicle)
...
+        local pumpObject = data.entity
+        local vehicle = GetPlayersLastVehicle()
+
+        if vehicle == 0 or #(GetEntityCoords(vehicle) - GetEntityCoords(pumpObject)) > 3 then
+            return notify('Vehicle far from pump')
+        end
+        StartFueling(vehicle)
+    end)
+
+    exports.qtarget:AddTargetModel(ox.pumpModels, {
```
Comment (LukeWasTakenn) [inline client.lua:193]: Would probably want to add the export to all of the vehicles instead of all the pump models

File: client.lua [hunk 2]
```diff
@@ -179,6 +179,30 @@ local function StartFueling(vehicle)
...
+        options = {
+            {
+                event = 'ox_fuel:refuelVehicle',
+                icon = 'fas fa-gas-pump',
+                label = "Refuel vehicle",
+                canInteract = function() return not isFueling end
```
Comment (LukeWasTakenn) [inline client.lua:199]: If adding the export to all vehicles would need a check if we are inside the gas station zone

**Comment by LukeWasTakenn** [comment]
We are also looking at improving performance by redoing the way we look for pumps so qtarget support is not needed yet imo.

Probably best to get the core mechanics working first anyways.

File: client.lua (hunk 1 above)
Comment (DoluTattoo) [reply client.lua:193]: You mean enum all the vehicles somewhere in ox_fuel resources ? Sounds a pain for servers who use add-on vehicle, no?

Comment (LukeWasTakenn) [reply client.lua:193]: No I mean using the Vehicle export, adding support for gas cans would be pain otherwise

**Pull request closed without merging**